source/BoardingPanel.cpp

```diff
@@ -333,26 +331,64 @@ bool BoardingPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &command,
 				if(!yourCrew || !enemyCrew)
 					break;
 
-				// Your chance of winning this round is equal to the ratio of
-				// your power to the enemy's power.
-				double yourPower = (youAttack ?
-					attackOdds.AttackerPower(yourCrew) : defenseOdds.DefenderPower(yourCrew));
-				double enemyPower = (enemyAttacks ?
-					defenseOdds.AttackerPower(enemyCrew) : attackOdds.DefenderPower(enemyCrew));
+				if(youAttack)
+				{
+					// Your chance of winning this round is equal to the ratio of
+					// your power to the enemy's power.
+					double yourAttackPower = attackOdds.AttackerPower(yourCrew);
+					double total = yourAttackPower + attackOdds.DefenderPower(enemyCrew);
 
-				double total = yourPower + enemyPower;
-				if(!total)
-					break;
+					if(total)
+					{
+						if(Random::Real() * total >= yourAttackPower)
+						{
+							++yourCasualties;
+							you->AddCrew(-1);
+							if(!you->Crew())
```

Comment:
If you attack, should we end the combat when only First Last is left?

Reply:
I don't see why; if only First Last is left, then First Last might still want to attack alone.

Reply:
With the current mechanics, the game won't allow you to try and attack alone. 
If we allowed that after this PR, it would be a departure from the current implementation.

Which isn't to say we can't do it, simply that it'd be changing the way it currently works.

Reply:
> With the current mechanics, the game won't allow you to try and attack alone.

I checked a bit further, and the game indeed doesn't allow First Last to start an attack alone. That check exists on line 307, both before this PR and after it.
The second check (on line 326, also still in this PR) is in the number of combat rounds, which limits the maximum loss of crew per executed attack to 1/5th of the crew (and to 1 once First Last has 4 crew or less).
Those checks together seem to prevent First Last from dying when the other side doesn't counter-attack, and that mechanism still seems to work after this PR.

Reply:
Currently, the game limits you to 1 crew loss per 'click on the button', while the new implementation seems to put the limit at 'one crew per attack', which may kill 2 of your crew if you both attack. This allows for FL to be killed when attacking while you have other crew members left, which isn't possible in the current implementation.

Reply:
Good point, I changed the check to break off the attack when the player-character is alone. (Player will likely still die in the next round if the enemy keeps on attacking.)